### ISSUE TITLE
PM2: Keymetrics agent memory leak

### ISSUE THREAD

**Issue by mongris** [Unitech/pm2#2015]
My server was out of response last week because the memory is exhausted. So I am checking the top memory usage processes everyday  after last reboot.  And I got the suspect:

The first day:
![iv 8762r1h wf qg2r6xs](https://cloud.githubusercontent.com/assets/259816/13730995/2f59c95e-e9c4-11e5-8ed7-4fd99ad686b8.jpg)

The second day:
![gruf s7b fj3frzx6 vxn70](https://cloud.githubusercontent.com/assets/259816/13730997/3510274e-e9c4-11e5-9326-719e36392b9b.jpg)

The third day: 
![h95 w1d 271e zs 2 q7 k](https://cloud.githubusercontent.com/assets/259816/13730999/42092176-e9c4-11e5-9f8f-3fa02c4e0729.jpg)

The fourth day:
![2b16de4b-8b09-4889-a668-b67fa2443cbc](https://cloud.githubusercontent.com/assets/259816/13731007/63f789c6-e9c4-11e5-9884-87893a599542.jpg)

**Comment by Unitech** [comment]
Checked on 8 servers running the exact same PM2 version, no memory leak at all.

Could you copy/paste here ~/.pm2/agent.log?

If nothing interesting in that file, can you do:

```
$ DEBUG="*" pm2 link
```

(This will restart the agent in debug mode)

**Comment by mongris** [comment]
Thank you for the quick reply. It looks that there has some connection problem.

`
[Keymetrics.io] Launching agent
2016-03-09 09:43:00: Launching Interactor exposure
2016-03-09 09:43:03: [WATCHDOG] Launching
2016-03-09 09:43:03: [REV] Connecting to cassiope.keymetrics.io:43554
2016-03-09 09:43:03: [WATCHDOG] Connected to PM2
2016-03-09 09:43:03: [PUSH] Connected to local PM2
2016-03-09 09:43:03: [REV] Connected to PM2
2016-03-09 09:43:03: [REV] Connected to cassiope.keymetrics.io:43554
2016-03-09 09:43:03: [PUSH] Connected via TCP 80
2016-03-09 12:37:10: [object Object]
2016-03-09 12:58:48: New remote action open files triggered for process 7
2016-03-09 12:58:48: [REVERSE INTERACTOR] Message received from AXM for proc_id : 7 and action name open files
2016-03-09 12:59:10: New remote action open ports triggered for process 7
2016-03-09 12:59:10: [REVERSE INTERACTOR] Message received from AXM for proc_id : 7 and action name open ports
2016-03-09 12:59:22: New remote action uptime triggered for process 7
2016-03-09 12:59:22: [REVERSE INTERACTOR] Message received from AXM for proc_id : 7 and action name uptime
2016-03-09 12:59:38: New remote action top mem consuming triggered for process 7
2016-03-09 12:59:38: [REVERSE INTERACTOR] Message received from AXM for proc_id : 7 and action name top mem consuming
2016-03-09 14:16:10: [object Object]
2016-03-09 14:25:10: [object Object]
2016-03-09 16:24:47: Password does not match
2016-03-09 16:24:47: Password does not match
2016-03-09 17:40:10: [object Object]
2016-03-09 21:31:11: [object Object]
2016-03-09 21:55:11: [object Object]
2016-03-09 22:25:11: [object Object]
2016-03-10 02:16:11: [object Object]
2016-03-10 04:46:11: [object Object]
2016-03-10 18:26:11: New remote action top mem consuming triggered for process 7
2016-03-10 18:26:11: [REVERSE INTERACTOR] Message received from AXM for proc_id : 7 and action name top mem consuming
2016-03-10 21:28:13: [object Object]
2016-03-10 21:55:13: [object Object]
2016-03-10 22:01:13: [object Object]
2016-03-10 23:46:13: [object Object]
2016-03-10 23:47:08: Error: socket hang up
2016-03-10 23:49:13: [object Object]
2016-03-10 23:55:13: [object Object]
2016-03-10 23:58:13: [object Object]
2016-03-11 00:07:13: [object Object]
2016-03-11 00:19:13: [object Object]
2016-03-11 01:43:13: [object Object]
2016-03-11 01:58:13: [object Object]
2016-03-11 01:58:39: Error: read ETIMEDOUT
2016-03-11 02:25:13: [object Object]
2016-03-11 03:19:13: [object Object]
2016-03-11 04:22:13: [object Object]
2016-03-11 04:37:13: [object Object]
2016-03-11 04:55:13: [object Object]
2016-03-11 05:01:13: [object Object]
2016-03-11 05:40:13: [object Object]
2016-03-11 08:46:35: [object Object]
2016-03-11 09:27:21: [object Object]
2016-03-11 09:36:46: [object Object]
2016-03-11 09:53:49: [object Object]
2016-03-11 10:11:31: [object Object]
2016-03-11 10:16:40: [object Object]
2016-03-11 10:20:26: Error: socket hang up
2016-03-11 10:20:26: Error: socket hang up
2016-03-11 10:31:14: [object Object]
2016-03-11 10:41:24: [object Object]
2016-03-11 10:58:45: [object Object]
2016-03-11 10:59:52: Error: socket hang up
[Keymetrics.io] Launching agent
2016-03-11 11:23:39: Launching Interactor exposure
2016-03-11 11:23:40: [WATCHDOG] Launching
2016-03-11 11:23:40: [REV] Connecting to cassiope.keymetrics.io:43554
2016-03-11 11:23:40: [WATCHDOG] Connected to PM2
2016-03-11 11:23:40: [PUSH] Connected to local PM2
2016-03-11 11:23:40: [REV] Connected to PM2
2016-03-11 11:23:41: [PUSH] Connected via TCP 80
2016-03-11 11:23:41: [REV] Connected to cassiope.keymetrics.io:43554
2016-03-11 22:23:49: [object Object]
2016-03-11 22:59:49: [object Object]
2016-03-12 01:53:49: [object Object]
2016-03-12 01:56:49: [object Object]
2016-03-12 02:11:49: [object Object]
2016-03-12 02:14:49: [object Object]
2016-03-12 02:16:50: Error: connect ETIMEDOUT 62.210.94.153:443
2016-03-12 02:17:49: [object Object]
2016-03-12 02:19:14: Error: socket hang up
2016-03-12 02:23:49: [object Object]
2016-03-12 02:26:49: [object Object]
2016-03-12 02:32:49: [object Object]
2016-03-12 02:35:49: [object Object]
2016-03-12 02:38:49: [object Object]
2016-03-12 02:44:49: [object Object]
2016-03-12 02:47:49: [object Object]
2016-03-12 02:50:47: [PUSH] Connected via TCP 80
2016-03-12 02:53:49: [object Object]
2016-03-12 02:59:49: [object Object]
2016-03-12 03:05:49: [object Object]
2016-03-12 03:08:49: [object Object]
2016-03-12 03:09:44: [PUSH] Connected via TCP 80
2016-03-12 03:11:49: [object Object]
2016-03-12 03:14:49: [object Object]
2016-03-12 03:17:49: [object Object]
2016-03-12 03:20:49: [object Object]
2016-03-12 03:22:50: Error: connect ETIMEDOUT 62.210.94.153:443
2016-03-12 03:29:49: [object Object]
2016-03-12 03:29:55: [REV] Error Error: read ECONNRESET
    at exports._errnoException (util.js:874:11)
    at TCP.onread (net.js:544:26)
2016-03-12 03:29:55: [REV] Connection closed
2016-03-12 03:29:57: [REV] Retrying to connect cassiope.keymetrics.io:43554
2016-03-12 03:29:58: [REV] Connected to cassiope.keymetrics.io:43554
2016-03-12 03:30:44: Error: socket hang up
2016-03-12 03:41:49: [object Object]
2016-03-12 03:46:53: [REV] Error Error: read ECONNRESET
    at exports._errnoException (util.js:874:11)
    at TCP.onread (net.js:544:26)
2016-03-12 03:46:53: [REV] Connection closed
2016-03-12 03:46:55: [REV] Retrying to connect cassiope.keymetrics.io:43554
2016-03-12 03:46:56: [REV] Connected to cassiope.keymetrics.io:43554
2016-03-12 03:53:49: [object Object]
2016-03-12 04:14:32: [PUSH] Connected via TCP 80
2016-03-12 04:14:49: [object Object]
2016-03-12 04:17:49: [object Object]
2016-03-12 04:19:50: Error: connect ETIMEDOUT 62.210.94.153:443
2016-03-12 04:23:49: [object Object]
2016-03-12 04:29:49: [object Object]
2016-03-12 04:30:35: [PUSH] Connected via TCP 80
2016-03-12 04:35:49: [object Object]
2016-03-12 04:38:49: [object Object]
2016-03-12 04:40:51: Error: connect ETIMEDOUT 62.210.94.153:443
2016-03-12 04:41:49: [object Object]
2016-03-12 04:56:49: [object Object]
2016-03-12 04:59:49: [object Object]
2016-03-12 05:00:46: Error: socket hang up
2016-03-12 05:05:49: [object Object]
2016-03-12 05:08:49: [object Object]
2016-03-12 05:11:49: [object Object]
2016-03-12 05:14:49: [object Object]
2016-03-12 05:20:49: [object Object]
2016-03-12 05:26:49: [object Object]
2016-03-12 05:29:49: [object Object]
2016-03-12 05:31:50: Error: connect ETIMEDOUT 62.210.94.153:443
2016-03-12 05:35:49: [object Object]
2016-03-12 05:41:49: [object Object]
2016-03-12 05:47:49: [object Object]
2016-03-12 05:53:50: [object Object]
2016-03-12 05:56:50: [object Object]
2016-03-12 06:02:50: [object Object]
2016-03-12 06:14:50: [object Object]
2016-03-12 06:17:50: [object Object]
2016-03-12 06:26:50: [object Object]
2016-03-12 06:29:50: [object Object]
2016-03-12 06:35:50: [object Object]
2016-03-12 06:38:50: [object Object]
2016-03-12 06:47:50: [object Object]
2016-03-12 06:50:50: [object Object]
2016-03-12 06:51:43: Error: socket hang up
2016-03-12 06:53:50: [object Object]
2016-03-12 06:59:50: [object Object]
2016-03-12 07:08:50: [object Object]
2016-03-12 07:11:50: [object Object]
2016-03-12 07:14:50: [object Object]
2016-03-12 07:17:50: [object Object]
2016-03-12 07:20:50: [object Object]
2016-03-12 07:29:50: [object Object]
2016-03-12 07:30:51: Error: socket hang up
2016-03-12 07:32:50: [object Object]
2016-03-12 07:35:50: [object Object]
2016-03-12 07:41:50: [object Object]
2016-03-12 07:43:50: Error: connect ETIMEDOUT 62.210.94.153:443
2016-03-12 07:44:50: [object Object]
2016-03-12 07:47:50: [object Object]
2016-03-12 07:47:50: Error: connect EHOSTUNREACH 62.210.94.153:443
2016-03-12 07:50:50: [object Object]
2016-03-12 07:50:50: Error: connect EHOSTUNREACH 62.210.94.153:443
2016-03-12 07:53:46: Error: connect EHOSTUNREACH 62.210.94.153:443
2016-03-12 07:56:50: [object Object]
2016-03-12 08:02:50: [object Object]
2016-03-12 08:04:47: Error: socket hang up
2016-03-12 08:05:50: [object Object]
2016-03-12 08:17:50: [object Object]
2016-03-12 08:23:50: [object Object]
2016-03-12 08:26:50: [object Object]
2016-03-12 08:35:50: [object Object]
2016-03-12 08:38:50: [object Object]
2016-03-12 08:47:50: [object Object]
2016-03-12 08:56:50: [object Object]
2016-03-12 08:59:50: [object Object]
2016-03-12 09:02:50: [object Object]
2016-03-12 09:05:50: [object Object]
2016-03-12 09:07:50: Error: connect ETIMEDOUT 62.210.94.153:443
2016-03-12 09:08:50: [object Object]
2016-03-12 09:23:50: [object Object]
2016-03-12 14:32:50: [object Object]
2016-03-13 02:47:51: [object Object]
2016-03-13 02:56:51: [object Object]
2016-03-13 03:05:51: [object Object]
2016-03-13 03:17:51: [object Object]
2016-03-13 03:23:51: [object Object]
2016-03-13 03:38:51: [object Object]
2016-03-13 03:41:51: [object Object]
2016-03-13 03:44:51: [object Object]
2016-03-13 03:45:46: Error: socket hang up
2016-03-13 03:53:51: [object Object]
2016-03-13 03:59:51: [object Object]
2016-03-13 04:00:46: Error: socket hang up
2016-03-13 04:11:51: [object Object]
2016-03-13 04:23:51: [object Object]
2016-03-13 04:38:51: [object Object]
2016-03-13 04:59:51: [object Object]
2016-03-13 05:08:51: [object Object]
2016-03-13 05:14:51: [object Object]
2016-03-13 05:29:51: [object Object]
2016-03-13 05:35:51: [object Object]
2016-03-13 05:44:51: [object Object]
2016-03-13 05:56:51: [object Object]
2016-03-13 06:14:51: [object Object]
2016-03-13 06:20:51: [object Object]
2016-03-13 06:23:51: [object Object]
2016-03-13 06:29:51: [object Object]
2016-03-13 06:35:51: [object Object]
2016-03-13 06:38:51: [object Object]
2016-03-13 06:50:51: [object Object]
2016-03-13 07:11:51: [object Object]
2016-03-13 07:35:51: [object Object]
2016-03-13 07:53:51: [object Object]
2016-03-13 07:56:51: [object Object]
2016-03-13 08:11:51: [object Object]
2016-03-13 08:14:51: [object Object]
2016-03-13 08:23:51: [object Object]
2016-03-13 08:35:51: [object Object]
2016-03-13 08:56:51: [object Object]
2016-03-13 09:02:51: [object Object]
2016-03-13 09:11:51: [object Object]
2016-03-13 09:17:51: [object Object]
2016-03-13 09:20:51: [object Object]
2016-03-13 09:23:51: [object Object]
2016-03-13 09:35:32: [PUSH] Connected via TCP 80
[Keymetrics.io] Launching agent
2016-03-13 09:47:20: Launching Interactor exposure
2016-03-13 09:47:22: [WATCHDOG] Launching
2016-03-13 09:47:22: [REV] Connecting to cassiope.keymetrics.io:43554
2016-03-13 09:47:22: [WATCHDOG] Connected to PM2
2016-03-13 09:47:22: [PUSH] Connected to local PM2
2016-03-13 09:47:22: [REV] Connected to PM2
2016-03-13 09:47:22: [PUSH] Connected via TCP 80
2016-03-13 09:47:22: [REV] Connected to cassiope.keymetrics.io:43554
2016-03-13 15:48:47: Killing interactor
2016-03-13 15:48:47: Exiting Interactor
`

**Comment by mongris** [comment]
I have add a cron job to restart agent daily. Any suggestion more?

**Comment by Unitech** [comment]
How many applications do you manage with PM2? I'm inspecting, it looks like it is a problem with TCP data buffering

**Comment by Unitech** [comment]
Could you have a try of the development version: https://github.com/Unitech/pm2/commit/532d79f859e379958f1795524aae48e33fc6b359

```
$ npm install Unitech/pm2#development -g
$ pm2 update
```

https://nodejs.org/api/net.html#net_socket_buffersize

**Comment by mongris** [comment]
I use PM2 to manage 5 applications (cluster in 7 instances). I have updated to #development last commit, will update memory stat later. Thank you.

**Comment by Unitech** [comment]
I deployed the PM2#dev on 5 prod servers, it works like a charm, the agent memory has been almost divided by two

**Comment by Unitech** [comment]
![screenshot from 2016-03-27 14-32-30](https://cloud.githubusercontent.com/assets/757747/14065424/e942add6-f428-11e5-8b6c-9db7dbec33cc.png)

**Comment by mongris** [comment]
All of my applications relaunch in loop after update pm2 ... I dont know what's happened ...

**Comment by Unitech** [comment]
logs?

**Comment by Unitech** [comment]
```
$ pm2 save
$ pm2 kill
$ pkill -f node
$ pm2 resurrect
```

**Comment by mongris** [comment]
After restart my server and reupdate pm2, everything come in peace now ... Thank you! I will keep an eye on the memory stat.

**Comment by mongris** [comment]
<img width="666" alt="screen shot 2016-03-28 at 11 10 58" src="https://cloud.githubusercontent.com/assets/259816/14068163/109dec02-f4d7-11e5-83e0-be7935758b05.png">

The problem is not fixed, and drains the memory faster.

**Comment by Unitech** [comment]
In the ~/.pm2/agent.log do you have [CRITICAL] log messages?

Could you send me an email with your Bucket name to alexandre [AT] keymetrics [DOT] io

Thank you

**Comment by mongris** [comment]
There has NO [CRITICAL] log msg:

```
2016-03-26 23:54:12: [REV] Retrying to connect cassiope.keymetrics.io:43554
2016-03-26 23:54:13: [REV] Error Error: connect ECONNREFUSED 195.154.156.78:43554
    at Object.exports._errnoException (util.js:874:11)
    at exports._exceptionWithHostPort (util.js:897:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1063:14)
2016-03-26 23:54:13: [REV] Connection closed
2016-03-26 23:54:13: [PUSH] Connected via TCP 80
2016-03-26 23:54:15: [REV] Retrying to connect cassiope.keymetrics.io:43554
2016-03-26 23:54:15: [REV] Connected to cassiope.keymetrics.io:43554
2016-03-27 01:18:11: [object Object]
2016-03-27 02:03:05: [PUSH] Connected via TCP 80
2016-03-27 02:03:08: [REV] Connection closed
2016-03-27 02:03:10: [REV] Retrying to connect cassiope.keymetrics.io:43554
2016-03-27 02:03:11: [REV] Error Error: connect ECONNREFUSED 195.154.156.78:43554
    at Object.exports._errnoException (util.js:874:11)
    at exports._exceptionWithHostPort (util.js:897:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1063:14)
2016-03-27 02:03:11: [REV] Connection closed
2016-03-27 02:03:13: [REV] Retrying to connect cassiope.keymetrics.io:43554
2016-03-27 02:03:13: [REV] Error Error: connect ECONNREFUSED 195.154.156.78:43554
    at Object.exports._errnoException (util.js:874:11)
    at exports._exceptionWithHostPort (util.js:897:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1063:14)
2016-03-27 02:03:13: [REV] Connection closed
2016-03-27 02:03:15: [REV] Retrying to connect cassiope.keymetrics.io:43554
2016-03-27 02:03:15: [REV] Connected to cassiope.keymetrics.io:43554
2016-03-27 02:03:16: [PUSH] Connected via TCP 80
2016-03-27 02:06:33: [PUSH] Connected via TCP 80
2016-03-27 02:06:36: [REV] Connection closed
2016-03-27 02:06:38: [REV] Retrying to connect cassiope.keymetrics.io:43554
2016-03-27 02:06:38: [REV] Error Error: connect ECONNREFUSED 195.154.156.78:43554
    at Object.exports._errnoException (util.js:874:11)
    at exports._exceptionWithHostPort (util.js:897:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1063:14)
2016-03-27 02:06:38: [REV] Connection closed
2016-03-27 02:06:40: [REV] Retrying to connect cassiope.keymetrics.io:43554
2016-03-27 02:06:41: [REV] Error Error: connect ECONNREFUSED 195.154.156.78:43554
    at Object.exports._errnoException (util.js:874:11)
    at exports._exceptionWithHostPort (util.js:897:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1063:14)
2016-03-27 02:06:41: [REV] Connection closed
2016-03-27 02:06:43: [REV] Retrying to connect cassiope.keymetrics.io:43554
2016-03-27 02:06:43: [PUSH] Connected via TCP 80
2016-03-27 02:06:44: [REV] Connected to cassiope.keymetrics.io:43554
2016-03-27 02:08:58: [PUSH] Connected via TCP 80
2016-03-27 02:09:02: [REV] Connection closed
2016-03-27 02:09:04: [REV] Retrying to connect cassiope.keymetrics.io:43554
2016-03-27 02:09:05: [REV] Error Error: connect ECONNREFUSED 195.154.156.78:43554
    at Object.exports._errnoException (util.js:874:11)
    at exports._exceptionWithHostPort (util.js:897:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1063:14)
2016-03-27 02:09:05: [REV] Connection closed
2016-03-27 02:09:07: [REV] Retrying to connect cassiope.keymetrics.io:43554
2016-03-27 02:09:07: [REV] Error Error: connect ECONNREFUSED 195.154.156.78:43554
    at Object.exports._errnoException (util.js:874:11)
    at exports._exceptionWithHostPort (util.js:897:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1063:14)
2016-03-27 02:09:07: [REV] Connection closed
2016-03-27 02:09:09: [PUSH] Connected via TCP 80
2016-03-27 02:09:09: [REV] Retrying to connect cassiope.keymetrics.io:43554
2016-03-27 02:09:10: [REV] Connected to cassiope.keymetrics.io:43554
2016-03-27 04:27:11: [object Object]
2016-03-27 05:00:02: Killing interactor
2016-03-27 05:00:02: Exiting Interactor
[Keymetrics.io] Launching agent
2016-03-27 05:00:02: Launching Interactor exposure
2016-03-27 05:00:04: [WATCHDOG] Launching
2016-03-27 05:00:04: [REV] Connecting to cassiope.keymetrics.io:43554
2016-03-27 05:00:04: [WATCHDOG] Connected to PM2
2016-03-27 05:00:04: [PUSH] Connected to local PM2
2016-03-27 05:00:04: [REV] Connected to PM2
2016-03-27 05:00:04: [PUSH] Connected via TCP 80
2016-03-27 05:00:04: [REV] Connected to cassiope.keymetrics.io:43554
2016-03-27 19:33:05: [REV] Connection closed
2016-03-27 19:33:07: [REV] Retrying to connect cassiope.keymetrics.io:43554
2016-03-27 19:33:08: [REV] Error Error: connect ECONNREFUSED 195.154.156.78:43554
    at Object.exports._errnoException (util.js:874:11)
    at exports._exceptionWithHostPort (util.js:897:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1063:14)
2016-03-27 19:33:08: [REV] Connection closed
2016-03-27 19:33:10: [REV] Retrying to connect cassiope.keymetrics.io:43554
2016-03-27 19:33:10: [REV] Error Error: connect ECONNREFUSED 195.154.156.78:43554
    at Object.exports._errnoException (util.js:874:11)
    at exports._exceptionWithHostPort (util.js:897:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1063:14)
2016-03-27 19:33:10: [REV] Connection closed
2016-03-27 19:33:12: [REV] Retrying to connect cassiope.keymetrics.io:43554
2016-03-27 19:33:12: [REV] Connected to cassiope.keymetrics.io:43554
2016-03-27 19:33:12: [PUSH] Connected via TCP 80
2016-03-27 20:20:10: Killing interactor
2016-03-27 20:20:10: Exiting Interactor
[Keymetrics.io] Launching agent
2016-03-27 20:20:11: Launching Interactor exposure
2016-03-27 20:20:13: [WATCHDOG] Launching
2016-03-27 20:20:13: [REV] Connecting to cassiope.keymetrics.io:43554
2016-03-27 20:20:13: [WATCHDOG] Connected to PM2
2016-03-27 20:20:13: [PUSH] Connected to local PM2
2016-03-27 20:20:13: [REV] Connected to PM2
2016-03-27 20:20:13: [PUSH] Connected via TCP 80
2016-03-27 20:20:13: [REV] Connected to cassiope.keymetrics.io:43554
2016-03-27 20:25:14: Killing interactor
2016-03-27 20:25:14: Exiting Interactor
[Keymetrics.io] Launching agent
2016-03-27 20:25:15: Launching Interactor exposure
2016-03-27 20:25:16: [WATCHDOG] Launching
2016-03-27 20:25:16: [REV] Connecting to cassiope.keymetrics.io:43554
2016-03-27 20:25:16: [WATCHDOG] Connected to PM2
2016-03-27 20:25:16: [PUSH] Connected to local PM2
2016-03-27 20:25:16: [REV] Connected to PM2
2016-03-27 20:25:17: [REV] Connected to cassiope.keymetrics.io:43554
2016-03-27 20:25:17: [PUSH] Connected via TCP 80
2016-03-27 20:32:27: Killing interactor
2016-03-27 20:32:27: Exiting Interactor
[Keymetrics.io] Launching agent
2016-03-27 20:32:27: Launching Interactor exposure
2016-03-27 20:32:29: [WATCHDOG] Launching
2016-03-27 20:32:29: [REV] Connecting to cassiope.keymetrics.io:43554
2016-03-27 20:32:29: [WATCHDOG] Connected to PM2
2016-03-27 20:32:29: [PUSH] Connected to local PM2
2016-03-27 20:32:29: [REV] Connected to PM2
2016-03-27 20:32:29: [PUSH] Connected via TCP 80
2016-03-27 20:32:29: [REV] Connected to cassiope.keymetrics.io:43554
2016-03-27 20:33:35: Killing interactor
2016-03-27 20:33:35: Exiting Interactor
[Keymetrics.io] Launching agent
2016-03-27 20:33:37: Launching Interactor exposure
2016-03-27 20:33:38: [WATCHDOG] Launching
2016-03-27 20:33:38: [REV] Connecting to cassiope.keymetrics.io:43554
2016-03-27 20:33:38: [WATCHDOG] Connected to PM2
2016-03-27 20:33:38: [PUSH] Connected to local PM2
2016-03-27 20:33:38: [REV] Connected to PM2
2016-03-27 20:33:38: Killing interactor
2016-03-27 20:33:38: Exiting Interactor
[Keymetrics.io] Launching agent
2016-03-27 20:34:02: Launching Interactor exposure
2016-03-27 20:34:04: [WATCHDOG] Launching
2016-03-27 20:34:04: [REV] Connecting to cassiope.keymetrics.io:43554
2016-03-27 20:34:04: [WATCHDOG] Connected to PM2
2016-03-27 20:34:04: [PUSH] Connected to local PM2
2016-03-27 20:34:04: [REV] Connected to PM2
2016-03-27 20:34:04: [PUSH] Connected via TCP 80
2016-03-27 20:34:04: [REV] Connected to cassiope.keymetrics.io:43554
2016-03-27 20:46:30: Killing interactor
2016-03-27 20:46:30: Exiting Interactor
[Keymetrics.io] Launching agent
2016-03-27 20:46:31: Launching Interactor exposure
2016-03-27 20:46:33: [WATCHDOG] Launching
2016-03-27 20:46:33: [REV] Connecting to cassiope.keymetrics.io:43554
2016-03-27 20:46:33: [WATCHDOG] Connected to PM2
2016-03-27 20:46:33: [PUSH] Connected to local PM2
2016-03-27 20:46:33: [REV] Connected to PM2
2016-03-27 20:46:33: [REV] Connected to cassiope.keymetrics.io:43554
2016-03-27 20:46:33: [PUSH] Connected via TCP 80
2016-03-27 20:52:54: New remote action disk usage triggered for process 7
2016-03-27 20:52:54: [REVERSE INTERACTOR] Message received from AXM for proc_id : 7 and action name disk usage
2016-03-27 20:53:06: New remote action who triggered for process 7
2016-03-27 20:53:06: [REVERSE INTERACTOR] Message received from AXM for proc_id : 7 and action name who
2016-03-27 20:53:14: New remote action uptime triggered for process 7
2016-03-27 20:53:14: [REVERSE INTERACTOR] Message received from AXM for proc_id : 7 and action name uptime
2016-03-27 20:53:22: New remote action open ports triggered for process 7
2016-03-27 20:53:22: [REVERSE INTERACTOR] Message received from AXM for proc_id : 7 and action name open ports
2016-03-27 20:53:50: New remote action top mem consuming triggered for process 7
2016-03-27 20:53:50: [REVERSE INTERACTOR] Message received from AXM for proc_id : 7 and action name top mem consuming
2016-03-27 21:52:40: [object Object]
2016-03-27 22:55:34: SyntaxError: Unexpected token <
2016-03-27 22:58:35: SyntaxError: Unexpected token <
2016-03-27 23:01:34: SyntaxError: Unexpected token <
2016-03-27 23:04:34: SyntaxError: Unexpected token <
2016-03-27 23:07:34: SyntaxError: Unexpected token <
2016-03-27 23:10:34: SyntaxError: Unexpected token <
2016-03-27 23:13:34: SyntaxError: Unexpected token <
2016-03-27 23:16:35: SyntaxError: Unexpected token <
2016-03-27 23:19:35: SyntaxError: Unexpected token <
2016-03-27 23:22:41: [object Object]
[Keymetrics.io] Launching agent
2016-03-28 06:15:34: Launching Interactor exposure
2016-03-28 06:15:35: [WATCHDOG] Launching
2016-03-28 06:15:35: [REV] Connecting to cassiope.keymetrics.io:43554
2016-03-28 06:15:35: [WATCHDOG] Connected to PM2
2016-03-28 06:15:35: [PUSH] Connected to local PM2
2016-03-28 06:15:35: [REV] Connected to PM2
2016-03-28 06:15:35: [REV] Connected to cassiope.keymetrics.io:43554
2016-03-28 06:15:35: [PUSH] Connected via TCP 80
2016-03-28 06:16:55: Killing interactor
2016-03-28 06:16:55: Exiting Interactor
[Keymetrics.io] Launching agent
2016-03-28 06:16:56: Launching Interactor exposure
2016-03-28 06:16:57: [WATCHDOG] Launching
2016-03-28 06:16:57: [REV] Connecting to cassiope.keymetrics.io:43554
2016-03-28 06:16:57: [WATCHDOG] Connected to PM2
2016-03-28 06:16:57: [PUSH] Connected to local PM2
2016-03-28 06:16:57: [REV] Connected to PM2
2016-03-28 06:16:57: [PUSH] Connected via TCP 80
2016-03-28 06:16:57: [REV] Connected to cassiope.keymetrics.io:43554
2016-03-28 06:30:52: New remote action disk usage triggered for process 7
2016-03-28 06:30:52: [REVERSE INTERACTOR] Message received from AXM for proc_id : 7 and action name disk usage
2016-03-28 06:31:00: New remote action top mem consuming triggered for process 7
2016-03-28 06:31:00: [REVERSE INTERACTOR] Message received from AXM for proc_id : 7 and action name top mem consuming
2016-03-28 06:31:33: New remote action top cpu consuming triggered for process 7
2016-03-28 06:31:33: [REVERSE INTERACTOR] Message received from AXM for proc_id : 7 and action name top cpu consuming
2016-03-28 06:55:53: [PUSH] Connected via TCP 80
2016-03-28 06:57:41: [PUSH] Connected via TCP 80
2016-03-28 06:57:46: [REV] Connection closed
2016-03-28 06:57:48: [REV] Retrying to connect cassiope.keymetrics.io:43554
2016-03-28 06:57:48: [REV] Error Error: connect ECONNREFUSED 195.154.156.78:43554
    at Object.exports._errnoException (util.js:874:11)
    at exports._exceptionWithHostPort (util.js:897:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1063:14)
2016-03-28 06:57:48: [REV] Connection closed
2016-03-28 06:57:50: [REV] Retrying to connect cassiope.keymetrics.io:43554
2016-03-28 06:57:51: [REV] Error Error: connect ECONNREFUSED 195.154.156.78:43554
    at Object.exports._errnoException (util.js:874:11)
    at exports._exceptionWithHostPort (util.js:897:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1063:14)
2016-03-28 06:57:51: [REV] Connection closed
2016-03-28 06:57:53: [REV] Retrying to connect cassiope.keymetrics.io:43554
2016-03-28 06:57:53: [REV] Connected to cassiope.keymetrics.io:43554
2016-03-28 06:57:54: [PUSH] Connected via TCP 80
2016-03-28 08:44:04: [object Object]
2016-03-28 11:50:04: [object Object]
2016-03-28 16:23:22: [PUSH] Connected via TCP 80
2016-03-28 16:23:25: [REV] Connection closed
2016-03-28 16:23:27: [REV] Retrying to connect cassiope.keymetrics.io:43554
2016-03-28 16:23:28: [REV] Error Error: connect ECONNREFUSED 195.154.156.78:43554
    at Object.exports._errnoException (util.js:874:11)
    at exports._exceptionWithHostPort (util.js:897:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1063:14)
2016-03-28 16:23:28: [REV] Connection closed
2016-03-28 16:23:30: [REV] Retrying to connect cassiope.keymetrics.io:43554
2016-03-28 16:23:30: [REV] Error Error: connect ECONNREFUSED 195.154.156.78:43554
    at Object.exports._errnoException (util.js:874:11)
    at exports._exceptionWithHostPort (util.js:897:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1063:14)
2016-03-28 16:23:30: [REV] Connection closed
2016-03-28 16:23:32: [REV] Retrying to connect cassiope.keymetrics.io:43554
2016-03-28 16:23:32: [REV] Connected to cassiope.keymetrics.io:43554
2016-03-28 16:23:33: [PUSH] Connected via TCP 80
2016-03-28 19:14:05: [object Object]
2016-03-28 20:59:05: [object Object]
2016-03-28 21:04:20: [REV] Connection closed
2016-03-28 21:04:22: [REV] Retrying to connect cassiope.keymetrics.io:43554
2016-03-28 21:04:22: [REV] Connected to cassiope.keymetrics.io:43554
2016-03-28 21:47:05: [object Object]
2016-03-29 02:14:05: [object Object]
2016-03-29 06:02:06: [object Object]
2016-03-29 06:03:34: Error: socket hang up
2016-03-29 07:56:44: [PUSH] Connected via TCP 80
2016-03-29 07:56:47: [REV] Connection closed
2016-03-29 07:56:49: [REV] Retrying to connect cassiope.keymetrics.io:43554
2016-03-29 07:56:49: [REV] Error Error: connect ECONNREFUSED 195.154.156.78:43554
    at Object.exports._errnoException (util.js:874:11)
    at exports._exceptionWithHostPort (util.js:897:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1063:14)
2016-03-29 07:56:49: [REV] Connection closed
2016-03-29 07:56:51: [REV] Retrying to connect cassiope.keymetrics.io:43554
2016-03-29 07:56:52: [REV] Error Error: connect ECONNREFUSED 195.154.156.78:43554
    at Object.exports._errnoException (util.js:874:11)
    at exports._exceptionWithHostPort (util.js:897:20)
    at TCPConnectWrap.afterConnect [as oncomplete] (net.js:1063:14)
2016-03-29 07:56:52: [REV] Connection closed
2016-03-29 07:56:52: [PUSH] Connected via TCP 80
2016-03-29 07:56:54: [REV] Retrying to connect cassiope.keymetrics.io:43554
2016-03-29 07:56:54: [REV] Connected to cassiope.keymetrics.io:43554

```

**Comment by Unitech** [comment]
Fixed on PM2 1.1.x
